src/ImageWrapper.cpp

```diff
@@ -59,6 +59,23 @@ extern "C" {
     return reinterpret_cast<Gosu_Image *>(image_data.get() ? new Gosu::Image(std::move(image_data)) : nullptr);
   }
 
+  void Gosu_Image_create_from_tiles(const char* source, int tile_width, int tile_height, Gosu_Image** buffer, int buffer_size, unsigned image_flags)
```

Comment:
@jlnr I implemented `Gosu_Image_create_from_tiles` by taking in and writing to a buffer, instead of returning an array.

I tried to use a `std::vector` however returning `images.data()` resulted in crashes, I assume were caused by going out of scope; Using the returned pointers would crash when accessing `Gosu::Image` functions. Some of the pointer addresses would be all or nearly all zeros.
If I ran `valgrind ruby ./gosu-examples` and loaded the Tutorial example it would work just fine...

I had something like this:
```cpp
void Gosu_Image_create_from_tiles(const char* source, int tile_width, int tile_height, unsigned image_flags)
{
  std::vector<Gosu::Image> gosu_images = Gosu::load_tiles(filename, width, height, image_flags);
  std::vector<Gosu_Image*> images;

  for (Gosu::Image img : gosu_images)
  {
    images.push_back(reinterpret_cast<Gosu_Image*>( new Gosu::Image(img) );
  }

  return images.data();
}
```

Reply:
Would it maybe be easier to have a function parameter that will be called one for every returned image with a context pointer, which it uses to add them all to the same array? Because your pasted code snippet returns a pointer to local data, and I don't think the updated function signature makes it clear how the number of created images will be returned.

Reply:
So, take in a function pointer that takes an `Gosu_Image*` and call it for each image?
e.g.
```cpp
void Gosu_Image_create_from_tiles(void* callback(Gosu_Image* ptr), const char* source, int tile_width, int tile_height, unsigned image_flags)
{
  std::vector<Gosu::Image> gosu_images = Gosu::load_tiles(filename, width, height, image_flags);

  for (Gosu::Image img : gosu_images)
  {
    callback( (reinterpret_cast<Gosu_Image*>( new Gosu::Image(img) );
  }
}
```

Reply:
Yep, except you probably need a context pointer again for Crystal. And making it a reference in the for loop can be a little more efficient: `for (Gosu::Image& image : gosu_images) ...`